dartagnan/src/main/java/com/dat3m/dartagnan/encoding/TypedValue.java

```diff
@@ -0,0 +1,27 @@
+package com.dat3m.dartagnan.encoding;
+
+import com.dat3m.dartagnan.expression.Type;
+import com.google.common.base.Preconditions;
+
+/*
+    TODO: This is a temporary solution and code should not be too dependent on this class.
+     The issue is as follows: while TypedFormulas allow us to construct SMT encodings agnostic to the actual
+     SMT types used in the encoding, the corresponding TypedValues generated from an SMT model
+     exposes the underlying SMT types.
+     What we rather want to have are Literal expressions, e.g., rather
+     than TypedValue<IntegerType, BigInteger>, we should return IntLiteral.
+     Similarly, rather than TypedValue<BooleanType, Boolean>, we should return BoolLiteral, etc.
```

Comment:
Why would we want the literal rather than the typed version? I understand that for literals the type information is obvious (because there is a unique SMT representation), but I guess the whole point is to have a unified interface

Reply:
The comment refers to literals in our IR, i.e., well-typed expressions. And we want to use them precisely because there is NO unique SMT representation even for literals: an IntLiteral may be represented by a BV formula or an Int formula and the user should not need to know this. TypedValue makes this SMT representation explicit, but since models of int formulas and bv formulas are both given by BigInteger, the fact that the SMT representation is leaked is not so obvious here.
However, suppose we encode aggregates as a single bit-string rather than a tuple, then we probably still want the IR model to make it look like it was a tuple all along.